### PR TITLE
dev/switchboard: Support stopping jobs (plus minor fixes)

### PR DESCRIPTION
**Pull request by max-cura**
This PR opens the following endpoint:
- `DELETE /api/v1/job/:id` to cancel a job

as previously, in #20, the switchboard was missing the code to assign the `stop_job:<JOB_ID>` permission.
Additionally, it makes some minor changes in the switchboard configuration, and updates the example configuration file to the new optional dev-mode-only SSL config.

### CODE REVIEW

**Comment by lschuermann** [comment]
For posterity's sake: I was slightly concerned whether `DELETE` is the appropriate HTTP verb here, because the resource will not actually vanish after this operation (a running job will just be stopped / canceled). However, looking up the specification for `DELETE` in RFC 7231, this seems to be fine:

```
[4.3.5](https://www.rfc-editor.org/rfc/rfc7231#section-4.3.5).  DELETE

   The DELETE method requests that the origin server remove the
   association between the target resource and its current
   functionality.  In effect, this method is similar to the rm command
   in UNIX: it expresses a deletion operation on the URI mapping of the
   origin server rather than an expectation that the previously
   associated information be deleted.

   If the target resource has one or more current representations, they
   might or might not be destroyed by the origin server, and the
   associated storage might or might not be reclaimed, depending
   entirely on the nature of the resource and its implementation by the
   origin server (which are beyond the scope of this specification).
   Likewise, other implementation aspects of a resource might need to be
   deactivated or archived as a result of a DELETE, such as database or
   gateway connections.  In general, it is assumed that the origin
   server will only allow DELETE on resources for which it has a
   prescribed mechanism for accomplishing the deletion.
 ```

File: switchboard/sql/SCHEMA.sql [hunk 1]
```diff
@@ -188,7 +188,9 @@ create type exit_status as enum (
     -- host reported success
     'host_terminated_with_success',
     -- job timeout
-    'host_terminated_timeout'
+    'host_terminated_timeout',
+    -- job was stopped by user
+    'stopped_by_user'
```
Comment (lschuermann) [inline switchboard/sql/SCHEMA.sql:193]: Can we call this cancelled? There should be a clear distinction between `host_terminated_with_success` (which is also a stop operation issued within the job's environment, either by a script or by a user), or this force-cancel operation?

**Review by lschuermann — CHANGES_REQUESTED** [review]
Looks good apart from the above comment.

File: switchboard/sql/SCHEMA.sql (hunk 1 above)
Comment (max-cura) [reply switchboard/sql/SCHEMA.sql:193]: I thought that it would be the supervisor closing the job on successful termination, not the switchboard?

**Review by lschuermann — APPROVED** [review]

**Pull request merged**